dattri/algorithm/data_shapley.py

```diff
@@ -0,0 +1,207 @@
+"""This module implement data shapley valuation."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from collections.abc import Callable
+    from typing import List, Optional, Tuple
+
+    from dattri.task import AttributionTask
+
+import torch
+from tqdm import tqdm
+
+from .base import BaseAttributor
+from .utils import _check_shuffle
+
+
+def default_dist_func(
+        batch_x: Tuple[torch.Tensor],
+        batch_y: Tuple[torch.Tensor],
+) -> torch.Tensor:
+    """Default distance function for KNN.
+
+    Args:
+        batch_x (Tuple[torch.Tensor]): The batch to be calculated
+            distances on. The embeddings is default to be the
+            first element.
+        batch_y (Tuple[torch.Tensor]): The batch to be calculated
+            distances on. The embeddings is default to be the
+            first element.
+
+    Returns:
+        Tensor: The calculated Euclidean distance.
+    """
+    coord1 = batch_x[0]
+    coord2 = batch_y[0]
+    return torch.cdist(coord1, coord2)
+
+
+class KNNShalpeyAttributor(BaseAttributor):
+    """KNN Data Shapley Attributer."""
+
+    def __init__(
+        self,
+        k_neighbors: int,
+        task: AttributionTask = None,
```

Comment:
Please add an issue to the issue page, where we can track the potential improvement. One design I am think is that maybe we can left `distance_func` visit the `task` and extract what ever they want.

```python
def distance_func(batch_x, batch_y, task=None):
    ...
```

Reply:
Please check issue #124